### PR TITLE
Decompile code_080023E4.s

### PR DESCRIPTION
**Pull request by Solitai7e**

### CODE REVIEW

File: src/code_080023A4.c [hunk 1]
```diff
@@ -1,12 +1,170 @@
...
+    return *(u8 *) (
+        *(u32 *) &(*gUnk_08D637AC[gUnk_089331AC[roomId].objectListIdx])[0].y
+        + arg1
+    );
+}
+
+void sub_080027A8(void)
+{
+    u32 var0 = 0;
+    CpuSet(&var0, gUnk_02023368, 0x05000008);
```
Comment (jiangzhengwenjz) [inline src/code_080023A4.c:163]: use `CpuFill32` instead

File: src/code_080023A4.c [hunk 2]
```diff
@@ -1,12 +1,170 @@
...
+}
+
+void sub_080027C8(void)
+{
+    u32 var0 = 0;
+    CpuSet(&var0, gUnk_02028BF0, 0x05000008);
```
Comment (jiangzhengwenjz) [inline src/code_080023A4.c:169]: use `CpuFill32` instead

File: src/code_080023A4.c [hunk 3]
```diff
@@ -1,12 +1,170 @@
...
+    for (;;) {
+        if (
+            var1->unk01 == var2 &&
+            var1->unk02 == x &&
+            var1->unk03 == y
+        ) {
+            return var1;
+        }
+
+        var1 = (void *) ((u32) var1 + var1->unk04);
```
Comment (jiangzhengwenjz) [inline src/code_080023A4.c:101]: use GNU Extension, `var1 = (void *)var1 + var1->unk04;`

File: src/code_080023A4.c [hunk 4]
```diff
@@ -1,12 +1,170 @@
...
+        if (
+            var1->unk01 == var2 &&
+            var1->unk02 == x &&
+            var1->unk03 == y &&
+            --arg3 == 0xFF
+        ) {
+            return var1;
+        }
+
+        var1 = (void *) ((u32) var1 + var1->unk04);
```
Comment (jiangzhengwenjz) [inline src/code_080023A4.c:121]: `var1 = (void *)var1 + var1->unk04;`

File: src/code_080023A4.c [hunk 5]
```diff
@@ -1,12 +1,170 @@
...
+}
+
+u8 sub_08002750(u16 roomId)
+{
+    return (*gUnk_08D637AC[gUnk_089331AC[roomId].objectListIdx])[0].unk0D;
+}
+
+u8 sub_08002774(u16 roomId, u8 arg1)
+{
+    return *(u8 *) (
```
Comment (jiangzhengwenjz) [inline src/code_080023A4.c:154]: same issue here

File: src/code_080023A4.c [hunk 6]
```diff
@@ -1,12 +1,170 @@
...
+}
+
+u8 sub_080026F8(u16 roomId)
+{
+    return (*gUnk_08D637AC[gUnk_089331AC[roomId].objectListIdx])[0].type;
+}
+
+u8 sub_0800271C(u16 roomId, u8 arg1)
+{
+    return *(u8 *) (
```
Comment (jiangzhengwenjz) [inline src/code_080023A4.c:141]: Maybe unk04 is a union? 
Also I don't see the point in casting to `u32 *`, given it's a `u8` array

File: src/code_080023A4.c [hunk 7]
```diff
@@ -1,12 +1,170 @@
 #include "code_080023A4.h"
-#include "data.h"
 
 // TODO: define file boundaries
```
Comment (jiangzhengwenjz) [inline src/code_080023A4.c:3]: still indefinite?

File: src/code_080023A4.c (hunk 6 above)
Comment (Solitai7e) [reply src/code_080023A4.c:141]: The `u32 *` cast I think is there so that it'd output an `ldr` instead of an `ldrb`, not sure.

Comment (jiangzhengwenjz) [reply src/code_080023A4.c:141]: oh yes. sorry i read it wrong

**Pull request merged**